File: 백준/Silver/11047. 동전 0/동전 0.cc
#include <stdio.h>

int main() {
    int a, b, i, cnt=0;
    int c[10];
    scanf("%d %d", &a, &b);
    i = a - 1;
    for(int i = 0; i<a; i++) { scanf("%d", &c[i]); }
    while (b > 0) {
        if (c[i] > b) { i--; }
        else { b = b - c[i]; cnt++; }
    }
    printf("%d", cnt);
    return 0;
}

File: 백준/Silver/11047. 동전 0/동전 0.js
let input = require("fs").readFileSync("/dev/stdin").toString().split("\n");
let [_, money] = input.shift().split(" ").map(Number);
input = input.sort((i, j) => i - j).map(Number);
let cnt = 0;

for (let i = input.length - 1; i > 0; i--) {
  cnt += Math.floor(money / input[i]);
  money = money % input[i];
}

console.log(cnt);